python/tank/path_cache.py

```diff
@@ -572,250 +565,417 @@ def _do_full_sync(self, cursor):
-        
-        log.debug("The following FilesystemLocation ids need replaying: %s" % created_folder_ids)
-        
+
+        # get the ids that are missing from shotgun
+        # need to use this weird special filter syntax
+        log.debug(
+            "Doing an incremental sync, so getting FilesystemLocation entries for "
+            "the following ids: %s" % created_folder_ids
+        )
+        # Retrieve all the newly created folders and rewire the result so it can be indexed by id.
+        created_folder_entities = self._get_filesystem_location_entites(created_folder_ids)
+        created_folder_entities = dict(
+            (entity["id"], entity) for entity in created_folder_entities
+        )
+
+        new_items = []
+
+        for event in sg_data:
+            sg_folder_ids = event["meta"].get("sg_folder_ids")
+
+            if event["event_type"] == "Toolkit_Folders_Delete":
+                # Remove all the entries associated with that event.
+                self._remove_filesystem_location_entities(cursor, sg_folder_ids)
+            elif event["event_type"] == "Toolkit_Folders_Create":
+                # For every folder in the create event.
+                for folder_id in sg_folder_ids:
+                    # If the entry is actually part of the end result, we'll add it!
+                    if folder_id in created_folder_entities:
+                        new_item = self._import_filesystem_location_entry(cursor, created_folder_entities[folder_id])
+                        # If the entry was actually imported.
+                        if new_item:
+                            new_items.append(new_item)
+
+        self._update_last_event_log_synced(cursor, max_event_log_id)
+
+        self._connection.commit()
+
         # run the actual sync - and at the end, inser the event_log_sync data marker
         # into the database to show where to start syncing from next time.
-        return self._replay_folder_entities(cursor, max_event_log_id, created_folder_ids)
+        return new_items
+
+    def _get_filesystem_location_entites(self, folder_ids):
+        """
+        Retrieves filesystem location entities from Shotgun.
 
+        :param list folder_ids: List of ids of entities to retrieve. If None, every entry is returned.
 
-    def _replay_folder_entities(self, cursor, max_event_log_id, ids=None):
+        :returns: List of FilesystemLocation entity dictionaries with keys:
+            - id
+            - type
+            - configuration_metadata
+            - is_primary
+            - linked_entity_id
+            - path
+            - linked_entity_type
+            - code
         """
-        Does the actual download from shotgun and pushes those changes
-        to the path cache. If ids is None, this indicates a full sync, and 
-        the path cache db table is cleared first. If not, the table
-        is appended to.
-        
+
+        if folder_ids:
```

Comment:
i am missing a debug log output before the call - we got a nice one after the call, can we add one before the find too? Especially when we are getting *all* fs locations, this is useful.